lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -125,14 +138,35 @@ export abstract class ClientApplication {
         this.eventHandler = new EventHandler(this.logger, this.browserCrypto);
 
         // Initialize the browser storage class.
-        this.browserStorage = this.isBrowserEnvironment ? 
-            new BrowserCacheManager(this.config.auth.clientId, this.config.cache, this.browserCrypto, this.logger) : 
+        this.browserStorage = this.isBrowserEnvironment ?
+            new BrowserCacheManager(this.config.auth.clientId, this.config.cache, this.browserCrypto, this.logger) :
             DEFAULT_BROWSER_CACHE_MANAGER(this.config.auth.clientId, this.logger);
 
         // Initialize the token cache
         this.tokenCache = new TokenCache(this.config, this.browserStorage, this.logger, this.browserCrypto);
     }
 
+    /**
+     * Initializer function to perform async startup tasks such as connecting to WAM extension
+     */
+    async initialize(): Promise<void> {
```

Comment:
Are we going to keep this separate from `handleRedirectPromise` for now? Or should we roll this together now?

Reply:
Might be best to keep separate for now, in v3 we can combine.

Reply:
I'd like to keep it separate for this PR but I'd like to roll it in eventually (whether that's in v3 or potentially sooner)